### PR TITLE
feat(loans): allow multiple physical copies per borrower (#238)

### PR DESCRIPTION
**Pull request by fabiodalez-dev**
## Summary

- add an opt-in Loans setting, disabled by default, for lending distinct physical copies of the same title to one borrower
- centralize borrower/title multiplicity rules while keeping pending requests, reservations, copyless rows, and NCIP strict
- preserve per-copy overlap guards, eligibility, capacity, and max-active-loan enforcement
- streamline the desk batch flow by retaining the selected title and focusing the next copy scan
- keep copy identity distinct through approval, reassignment, copy-loss recovery, and scheduled-loan activation
- seed and document the setting in all five shipped locales

Addresses the classroom workflow described in Discussion #238 while preserving the standard library behavior by default.

## Safety and upgrade behavior

- missing settings resolve to OFF, so existing installations are unchanged without a migration
- saving Loan settings lazily persists the new key; fresh installs receive an idempotent seed
- disabling the option never deletes existing loans and lifecycle jobs continue preserving distinct copy identity
- the Issue #366 overdue-copy and pickup-recovery invariants remain unchanged

## Verification

- `bash scripts/ci-quality-local.sh` with the dedicated E2E database: passed
- `php tests/multiple-copy-loans-238.unit.php`: 50/50 passed
- `php tests/issue-366-full-scenario.unit.php`: 49/49 passed
- `php tests/pickup-ready-copy-free-366.unit.php`: 8/8 passed
- `php tests/loan-edge-cases.unit.php`: 64/64 passed
- PHPStan level 5, locale parity, frontend lint, audits, schema/migration gates: passed
- browser QA: desktop/mobile setting layout, toggle persistence, contextual copy, and next-copy focus verified

## Summary by CodeRabbit

* **Nuove funzionalità**
  * Aggiunta un’opzione configurabile per consentire più prestiti attivi dello stesso titolo su copie fisiche diverse.
  * Migliorati i flussi di creazione, approvazione, riassegnazione e gestione automatica dei prestiti.
  * Aggiunti messaggi e istruzioni per la scansione consecutiva delle copie.

* **Correzioni**
  * Evitate assegnazioni di copie già impegnate dallo stesso utente.
  * Mantenuti i vincoli su prenotazioni, richieste, disponibilità e limiti di prestito.

* **Documentazione e localizzazione**
  * Aggiornate impostazioni, documentazione e traduzioni in più lingue.

### CODE REVIEW

**Comment by fabiodalez-dev** [comment]
@coderabbitai review

**Comment by fabiodalez-dev** [comment]
@coderabbitai review

File: app/Support/LoanMultiplicityPolicy.php [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+     */
+    public function hasBlockingLoan(
+        int $bookId,
+        int $userId,
+        bool $operationWillBindCopy,
+        ?int $excludeLoanId = null
+    ): bool {
+        $strict = !$operationWillBindCopy || !$this->isEnabled();
+        $excludeSql = $excludeLoanId !== null ? ' AND id <> ?' : '';
+        $activeCopyPredicate = $strict ? '' : ' AND copia_id IS NULL';
```
[the remark this answers is not in the document]
Comment (fabiodalez-dev) [reply app/Support/LoanMultiplicityPolicy.php]: Addressed in 40662cbf. The relaxed predicate now keeps promoted queue reservations blocking while they are `origine=prenotazione` and `prenotato`/`da_ritirare`, even with `copia_id`, while copy-bound direct physical loans remain compatible. Regression coverage verifies the blocking reservation and both direct scheduled states; the feature suite passes 43/43.

**Comment by fabiodalez-dev** [comment]
@coderabbitai review

File: app/Services/ReservationReassignmentService.php [hunk 2]
```diff
@@ -167,56 +138,155 @@ public function reassignOnNewCopy(int $libroId, int $newCopiaId): void
...
+                    }
+
+                    $startsBeforeCandidateEnds = (string) $commitment['data_prestito']
+                        <= (string) $candidate['data_scadenza'];
+                    $endsAfterCandidateStarts = $commitment['stato'] === 'in_ritardo'
+                        || (string) $commitment['data_scadenza'] >= (string) $candidate['data_prestito'];
+                    if ($startsBeforeCandidateEnds && $endsAfterCandidateStarts) {
+                        $hasOverlap = true;
+                    }
+                }
```
[the remark this answers is not in the document]
Comment (fabiodalez-dev) [reply app/Services/ReservationReassignmentService.php]: Addressed in d9436a64. `LoanMultiplicityPolicy` now owns the canonical open-commitment predicate and exposes one typed batch lock/query reused by both `committedCopyIds()` and `reassignOnNewCopy()`. The shared comparator centralizes same-borrower physical identity plus inclusive/open-ended overlap semantics. This preserves the book → loans → copies lock order and avoids N+1 queries. Dedicated coverage is now 50/50, with the full local quality mirror green.

**Comment by fabiodalez-dev** [comment]
Index follow-up from the latest review: I verified the existing schema and query plan before changing the core table. `prestiti` already has `idx_copia_id(copia_id)` and `idx_libro_utente(libro_id, utente_id)`; MySQL `EXPLAIN` selects `idx_copia_id` for both correlated anti-joins with an estimated single-row lookup per copy. The second anti-join does not filter by `utente_id`, so the suggested `(libro_id, utente_id, copia_id)` index would not help that path. Since copy-loss reassignment is infrequent and the retry ceiling is for contention rather than the normal query count, this PR intentionally avoids an unproven schema migration and extra write amplification. A compound-index change should be driven by a production-scale benchmark in a dedicated performance migration.

**Comment by fabiodalez-dev** [comment]
@coderabbitai review

**Comment by fabiodalez-dev** [comment]
Addressed the final CodeRabbit maintainability finding in 027dd2a4. `OPEN_ACTIVE_STATES_SQL` is now the single source for the active open-state list used by both `OPEN_COMMITMENT_PREDICATE` and `hasBlockingLoan()`, with SQL structure, precedence and bindings unchanged. The feature suite remains 50/50 and the complete local quality mirror passed again.

**Comment by fabiodalez-dev** [comment]
@coderabbitai review

File: tests/multiple-copy-loans-238.spec.js [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+async function deskCreate(page, borrower, inv, { saveAndNew = false } = {}) {
+  await page.goto(`${BASE}/admin/loans/create`);
+  await page.waitForLoadState('domcontentloaded');
+  await fillAutocomplete(page, '#utente_search', '#utente_suggest', borrower.surname, '/api/search/utenti');
+  await scanCopy(page, inv);
+  const btn = saveAndNew
+    ? 'button[name="save_and_new"]'
+    : 'button[type="submit"]:not([name="save_and_new"])';
+  await page.locator(btn).click();
+  return waitLoanOutcome(page);
```
[the remark this answers is not in the document]
Comment (fabiodalez-dev) [reply tests/multiple-copy-loans-238.spec.js:183]: Verificato sul flusso reale: il form app/Views/prestiti/crea_prestito.php non usa data-swal-confirm, non registra un handler submit con Swal.fire() e i due pulsanti eseguono direttamente POST e redirect. Le SweetAlert dei prestiti sono nelle azioni distruttive del partial loan-actions-swal.php, non nella creazione. Attendere o cliccare .swal2-confirm qui introdurrebbe un timeout. I submit restano correttamente sincronizzati su waitLoanOutcome(); il test feature e il flusso completo sono stati eseguiti senza popup. Risolvo come falso positivo verificato.
